framer/Layer.coffee

```diff
@@ -883,35 +885,82 @@ class exports.Layer extends BaseClass
 	##############################################################
 	## ANIMATION
 
-	animate: (options) ->
-		# console.warn "Layer.animate is deprecated: please use Layer.animateTo instead"
-		properties = options.properties
-		delete options.properties
-		@animateTo(properties, options)
-
-	animateTo: (properties,options={}) ->
-		_.defaults(options,properties.options)
+	animateToState: (stateName, options={}) ->
+		properties = @_stateMachine.switchTo stateName
+		if @_stateMachine.previousName is @_stateMachine.currentName
+			shouldChange = false
+			for property,value of properties
+				if @[property] != value
+					shouldChange = true
+					break
+			if not shouldChange
+				return null
+		finished = options.completion
+		options.completion = =>
+			@_stateMachine.emit(Events.StateDidSwitch, @_stateMachine.previousName, @_stateMachine.currentName, @)
+			finished?()
+		@animate properties, options
+
+	animate: (properties,options={}) ->
+		if typeof properties == 'string'
+			stateName = properties
+			return @animateToState stateName, options
+
+		#Support the old properties syntax
+		if properties.properties?
+			console.warn "Using Layer.animate with 'properties' key is deprecated: please provide properties directly and use the 'options' key to provide animation options instead"
+			options = properties
+			properties = options.properties
+			delete options.properties
+
+		_.defaults(options,properties.options,@animationOptions)
 		delete properties.options
-		options.properties = Animation.filterAnimatableProperties(properties)
-		options.layer = @
+
+		animatableProperties = Animation.filterAnimatableProperties(properties)
+		nonAnimatableProperties = _.omit(_.clone(properties),_.keys(animatableProperties))
 
 		start = options.start
 		start ?= true
 		delete options.start
-
-		if options.instant
+		instant = options.instant ? false
```

Comment:
It's not a ternary, but an existential operator: http://coffeescript.org/#operators